src/tensor_operations/matrix_decomposition.jl

```diff
@@ -581,8 +581,8 @@ using NDTensors: map_diag!
 function sqrt_decomp(D::ITensor, u::Index, v::Index)
   (storage(D) isa Union{Diag,DiagBlockSparse}) ||
     error("Must be a diagonal matrix ITensor.")
-  sqrtDL = diag_itensor(u, dag(u)')
-  sqrtDR = diag_itensor(v, dag(v)')
+  sqrtDL = adapt(datatype(D))(diag_itensor(u, dag(u)'))
+  sqrtDR = adapt(datatype(D))(diag_itensor(v, dag(v)'))
```

Comment:
```suggestion
  sqrtDL = adapt(datatype(D), diag_itensor(u, dag(u)'))
  sqrtDR = adapt(datatype(D), diag_itensor(v, dag(v)'))
```
Just a style thing, normally I would only use `adapt(datatype(D))` if I am passing it as an argument to another function, for example `map(adapt(datatype(D)), itensor_list)`.